docs/main/Help/migration-guide.md

```diff
@@ -0,0 +1,76 @@
+```typescript
+import Hero from '@ulixee/hero';
+
+async function run() {
+  const hero = new Hero();
+  await hero.goto('https://example.org/');
+  await hero.waitForPaintingStable();
+
+  const html = await hero.document.documentElement.outerHTML;
+  const title = await hero.document.title;
+  const intro = await hero.querySelector('p').textContent;
+  await hero.close();
+}
+
+run().catch(error => console.log(error));
+```
+
+## No more Default Agent
+
+If you were using the default "Agent" from `const agent = require('secret-agent')`, you'll need to move to a mode of constructing a Hero object for each session.
+
+## Server Setup
+
+If you use a Client/Server Setup, you'll find that we moved the `Server` module out of Hero, and into [@ulixee/server][server]. You get the same functionality out of the box - a Server you can install on a machine and your DOM, interactions and other commands all serialize seamlessly.
+
+## Handlers
+
+Hero doesn't have a Handler or Server directly in the project. [@ulixee/server][server] is the new approach for setting up a remote Server, and we've decided to go a new direction for our preferred deployment strategy. The new strategy is called [Databoxes][databox], and greatly improves efficiency and has a much simpler remote deployment (well, it will soon :).
+
+Databoxes focus on creating a remotely callable "function" that wraps a Hero script in a composable "unit". You can vary inputs and collect outputs, or retry a script from a new IP address without having to think much about how to do that yourself.
+
+If you want to keep a model where you are using "driving" a script by queuing up a local "Client" that executes remotely (ie, like a SecretAgent Handler). To achieve client-size load-balancing, we recommend migrating to a tool like [`p-queue`](https://github.com/sindresorhus/p-queue). We've included an example in the Hero repo ([here](https://github.com/ulixee/hero/blob/27e1966c636f47519ed5d1ccc22273c1215855c1/examples/ulixee.org.ts)).
+
+## BrowserEmulators and HumanEmulators
+
+If you created a custom BrowserEmulator or HumanEmulator, those "concepts" have been merged into a single type of Plugin called an [Unblocked Plugin][unblocked-plugin]. You can see our plugins ported to the new format [here](https://github.com/unblocked-web/unblocked/plugins). We're planning to break up the BrowserEmulator into a series of smaller plugins with the end vision of making it far simpler to add workarounds as you find the need to add bot-blocker evasions.
+
+## Puppet
+
+To that end, the library that was formally passed into "Plugins" and "Human/Browser" Emulators has been merged with some parts of SecretAgent Core to form the [Unblocked Agent][agent]. The Unblocked Agent is now under it's own development lifecycle and has been put into a much tighter loop with DoubleAgent (aka, detection) development. We envision Hero as the home of Scraper productivity tooling, and Unblocked as the home of the bot cat and mouse game.
```

Comment:
I don't think "Scraper" should be capitalized as a proper noun.

Reply:
Ok, will fix